### PR TITLE
[IA-3414] Updated jersey-common dependencies

### PR DESCRIPTION
**Pull request by n8landolt**
## Description

We have an old version of jersey-common that has a security vulnerability. Update to the newest version to fix this issue.

Notes:
- The dependency terra-resource-buffer-client-0.4.3-SNAPSHOT is using "javax.ws.rs.client.Client;" which is incompatible with the upgrade to "jakarta.ws.rs.client.Client"

### CODE REVIEW

**Review by ddietterich — COMMENTED** [review]
Some comments on this WIP

File: client/gradle/dependencies.gradle [hunk 1]
```diff
@@ -1,6 +1,6 @@
 dependencies {
   ext {
-    jerseyVersion = "2.32"
+    jerseyVersion = "3.0.5"
```
[the remark this answers is not in the document]
Comment (n8landolt) [reply client/gradle/dependencies.gradle:3]: The purpose of the story is to update vulnerable dependencies. Jersey 2.32 is vulnerable.

[the remark this answers is not in the document]
Comment (n8landolt) [reply client/gradle/dependencies.gradle:3]: https://mvnrepository.com/artifact/org.glassfish.jersey.core/jersey-client/2.23

File: client/gradle/dependencies.gradle [hunk 2]
```diff
@@ -10,6 +10,11 @@ dependencies {
   implementation group: "org.glassfish.jersey.media", name: "jersey-media-json-jackson", version: jerseyVersion
   implementation group: "org.glassfish.jersey.media", name: "jersey-media-multipart", version: jerseyVersion
   implementation group: "org.glassfish.jersey.connectors", name: "jersey-jdk-connector", version:jerseyVersion
+
+  implementation group: 'javax.annotation', name: 'javax.annotation-api', version: '1.3.2'
```
[the remark this answers is not in the document]
Comment (n8landolt) [reply client/gradle/dependencies.gradle:14]: Working on getting the code happy.

File: client/src/main/resources/swaggercodegen/libraries/jersey2/JSON.mustache [hunk 3]
```diff
@@ -0,0 +1,62 @@
+package {{invokerPackage}};
```
[the remark this answers is not in the document]
Comment (n8landolt) [reply client/src/main/resources/swaggercodegen/libraries/jersey2/JSON.mustache:1]: I added it to change line 20. The import was coming from javax.es.rs.ext which no longer exists with the updated packages.

File: integration/build.gradle [hunk 4]
```diff
@@ -73,6 +73,12 @@ dependencies {
     // Use Terra Common Library on client side to retry direct Sam calls
     implementation group: "bio.terra", name: "terra-common-lib", version: "${tclClient}"
 
+    //Annotations
+    implementation group: 'javax.annotation', name: 'javax.annotation-api', version: '1.3.2'
```
[the remark this answers is not in the document]
Comment (n8landolt) [reply integration/build.gradle:77]: Some annotations don't exist in Jakarta. Nullable is one of them.

**Comment by ddietterich** [comment]
Is this PR still active or can it be closed?

**Pull request closed without merging**